### PR TITLE
Update user test

### PR DESCRIPTION
**Pull request by jcoyne**
Reverts part of #563

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -42,116 +42,130 @@
...
+      it { is_expected.to be false }
     end
   end
 
-  describe 'is_webauth_admin?' do
-    it 'should be true for ADMIN_GROUPS' do
-      allow(subject).to receive(:webauth_groups).and_return(User::ADMIN_GROUPS)
-      expect(subject.is_webauth_admin?).to be true
+  describe '#is_webauth_admin?' do
+    subject { user.is_admin? }
```
Comment (jmartin-sul) [inline spec/models/user_spec.rb]: this doesn't actually seem to be testing `#is_webauth_admin?` anymore?

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -42,110 +42,130 @@
...
+    let(:user) { described_class.new }
+    before do
+      allow(user).to receive(:webauth_groups).and_return(groups)
     end
 
-    it 'should be true when impersonating other group membership' do
-      allow(subject).to receive(:webauth_groups).and_return(User::ADMIN_GROUPS)
-      allow(subject).to receive(:groups).and_return(['workgroup:dlss:not-admin'])
```
Comment (jmartin-sul) [inline spec/models/user_spec.rb]: this change seems to lose the point of this test:  that the user is only in a non-admin group according to `#groups` (presumably due to impersonation of another group list).  but they are actually an admin according to `#webauth_groups` (this is what allows them to keep the link to re-set the groups to impersonate, even when they are impersonating a non-super-user).

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -243,33 +263,46 @@
...
+    end
+
+    context 'when impersonating' do
+      before do
+        user.set_groups_to_impersonate(%w(workgroup:dlss:impersonatedgroup1 workgroup:dlss:impersonatedgroup2))
       end
-      it 'should return the groups by impersonation' do
-        impersonated_groups = %w(workgroup:dlss:impersonatedgroup1 workgroup:dlss:impersonatedgroup2)
-        @user.set_groups_to_impersonate(impersonated_groups)
-        expect(@user.groups).to eq(impersonated_groups)
```
Comment (jmartin-sul) [inline spec/models/user_spec.rb]: i like the more thorough tests for what shouldn't come through from the actual webauth groups, but why lose the simple test for whether `#groups` returns the impersonated group list?  e.g., why not `let(:groups_to_impersonate) { %w(workgroup:dlss:impersonatedgroup1 workgroup:dlss:impersonatedgroup2) }`, use that in the `user.set_groups_to_impersonate` call, and then have something like
```ruby
it 'returns the impersonated groups' do
  expect(subject).to eq groups_to_impersonate
end
```

File: spec/models/user_spec.rb (hunk 1 above)
Comment (jcoyne) [reply spec/models/user_spec.rb]: https://github.com/sul-dlss/argo/pull/1075

File: spec/models/user_spec.rb (hunk 2 above)
Comment (jcoyne) [reply spec/models/user_spec.rb]: I'm not sure I follow. Here I'm only setting `webauth_groups`, and we see that when the `webauth_groups` have the admin group, then `is_webauth_admin?` returns `true`. Maybe your concern is addressed by #1075

**Pull request merged**